File: tools/mo/openvino/tools/mo/front/mxnet/leaky_relu.py
# Copyright (C) 2018-2024 Intel Corporation
# SPDX-License-Identifier: Apache-2.0

from openvino.tools.mo.ops.activation_ops import Elu, LeakyReLU, ReLU
from openvino.tools.mo.ops.gelu import GeLUOP
from openvino.tools.mo.ops.prelu import PReLU
from openvino.tools.mo.front.extractor import FrontExtractorOp
from openvino.tools.mo.front.mxnet.extractors.utils import get_mxnet_layer_attrs
from openvino.tools.mo.utils.error import Error
from openvino.tools.mo.utils.utils import refer_to_faq_msg


class LeakyReLUFrontExtractor(FrontExtractorOp):
    op = 'LeakyReLU'
    enabled = True

    @classmethod
    def extract(cls, node):
        attrs = get_mxnet_layer_attrs(node.symbol_dict)
        act_type = attrs.str('act_type', 'leaky')
        if act_type == 'prelu':
            prelu_attrs = {'channel_shared': 1,
                           'filler_type': 'constant',
                           'filler_value': 0,
                           'min': 0,
                           'max': 1,
                           'mean': 0,
                           'std': 0,
                           'sparse': -1,
                           'variance_norm': "caffe.FillerParameter.FAN_IN"}
            PReLU.update_node_stat(node, prelu_attrs)
        elif act_type == 'elu':
            alpha = attrs.float('slope', 0.25)
            Elu.update_node_stat(node, {'alpha': alpha})
        elif act_type == 'leaky':
            negative_slope = attrs.float('slope', 0.25)
            if negative_slope == 0:
                ReLU.update_node_stat(node)
            else:
                LeakyReLU.update_node_stat(node, {'negative_slope': negative_slope})
        elif act_type == 'gelu':
            GeLUOP.update_node_stat(node, {'approximation_mode': 'erf'})
        else:
            raise Error(
                "Operation '{}' not supported. Please register it as custom op. " +
                refer_to_faq_msg(86),
                act_type)

        return LeakyReLUFrontExtractor.enabled

File: src/frontends/tensorflow_common/src/op/leaky_relu.cpp
// Copyright (C) 2018-2025 Intel Corporation
// SPDX-License-Identifier: Apache-2.0
//

#include "common_op_table.hpp"
#include "openvino/op/constant.hpp"
#include "openvino/op/prelu.hpp"
#include "utils.hpp"

using namespace std;
using namespace ov::op;

namespace ov {
namespace frontend {
namespace tensorflow {
namespace op {
ov::OutputVector translate_leaky_relu_op(const NodeContext& node) {
    default_op_checks(node, 1, {"LeakyRelu", "LEAKY_RELU"});
    auto features = node.get_input(0);
    auto alpha = node.get_attribute<float>("alpha", 0.2f);
    auto alpha_const = create_same_type_const_scalar<float>(features, alpha);
    auto leaky_relu = make_shared<v0::PRelu>(features, alpha_const);
    set_node_name(node.get_name(), leaky_relu);
    return {leaky_relu};
}
}  // namespace op
}  // namespace tensorflow
}  // namespace frontend
}  // namespace ov
